### DISCUSSION TITLE
BBN does not start openbox/desktop, black screen

### DISCUSSION THREAD

**Discussion by atchik6056** [bareboat-necessities/lysmarine_gen#355] [General]
Hello
BBN hangs on initialising desktop after startup. blank screen with blinking cursor. Because I have only Bluetooth key/mouse it is very difficult to  diagnose. The only what I can do is to insert SD to another computer, mount and make some changes to config.
I suspect it start hanging after I successfully updated signalk 2.1.0 to 2.3.0 as described https://github.com/SignalK/signalk-server with:
$ sudo npm install -g signalk-server
it was done successfully, signalk server restarted and started getting all data, then I reboot.
Is there something additional I should do? or it is totally unrelated (I would assume)
How to kick desktop to startup?

**Comment by atchik6056** [comment]
forgot, I also updated from hot fixes::
cd /home/user/add-ons
./hot-fixes-install.sh
and it installed just one library file, I think lib*m0 something, apparently it was added not to long time ago 
could it affect desktop/touchsceen initialisation?

**Comment by mgrouch** [comment]
Ctrl-Alt-F1 or Ctrl-Alt-F2
should give you terminal from which you can troubleshoot

**Reply by atchik6056** [reply]
I know that staff being with linux for more 20 years :-)
only bluetooth keyboard on board unfortunately trap...
but thank you, 
anything I can do mounting sd on another computer?

**Reply by mgrouch** [reply]
You can try ssh from another computer.

**Reply by atchik6056** [reply]
right, I managed to fix it connecting to another switch via ethernet port

root cause: 
installation of libwidevinecdm0 via ./hot-fixes-install.sh broke /usr/share/X11/xorg.conf.d/90-touchinput.conf. 
it is not BBN problem but rather waveshare touchscreen driver for https://www.waveshare.com/product/7inch-dsi-lcd-c.htm.
I have reported them few months ago but apparently they have not fixed it

workaround: 
if you run ./hot-fixes-install.sh you need to make sure that  /usr/share/X11/xorg.conf.d/90-touchinput.conf is ok following ALL steps described in your guide section 1.35.1. Touchscreen https://bareboat-necessities.github.io/my-bareboat/bareboat-os.html#_touchscreen

more details on the fix:

/var/log/apt/history.log
...
Start-Date: 2023-10-19  11:58:43
Commandline: apt-get install -y -q libwidevinecdm0
Requested-By: user (1000)
Install: libwidevinecdm0:arm64 (4.10.2252.0+3)
End-Date: 2023-10-19  11:58:47

/var/log/Xorg.0.log
...
[ 10386.211] (==) Using config directory: "/etc/X11/xorg.conf.d"
[ 10386.211] (==) Using system config directory "/usr/share/X11/xorg.conf.d"
[ 10386.211] Parse error on line 5 of section InputClass in file /usr/share/X11/xorg.conf.d/90-touchinput.conf
	"WS_xinchDSI_Touch_c" is not a valid keyword in this section.
[ 10386.212] (EE) Problem parsing the config file
[ 10386.212] (EE) Error parsing the config file
[ 10386.212] (EE) 
Fatal server error:
[ 10386.212] (EE) no screens found(EE) 
[ 10386.213] (EE) 
Please consult the The X.Org Foundation support 
	 at http://wiki.x.org
 for help. 
[ 10386.213] (EE) Please also check the log file at "/var/log/Xorg.0.log" for additional information.
[ 10386.213] (EE) 
[ 10386.214] (EE) Server terminated with error (1). Closing log file.

user@lysmarine:/var/log $ cat /usr/share/X11/xorg.conf.d/90-touchinput.conf
Section "InputClass"
    Identifier "calibration"
    Driver "evdev"
    MatchProduct "WS Capacitive TouchScreen
WS_xinchDSI_Touch_c
bcm2835 (i2c@7e804000)"
    MatchDevicePath "/dev/input/event*"
    Option "EmulateThirdButton" "1"
    Option "EmulateThirdButtonTimeout" "750"
    Option "EmulateThirdButtonMoveThreshold" "30"
EndSection

fixed to:

user@lysmarine:/var/log $ cat /usr/share/X11/xorg.conf.d/90-touchinput.conf
Section "InputClass"
    Identifier "calibration"
    Driver "evdev"
    MatchProduct "WS Capacitive TouchScreenWS_xinchDSI_Touch_cbcm2835 (i2c@7e804000)"
    MatchDevicePath "/dev/input/event*"
    Option "EmulateThirdButton" "1"
    Option "EmulateThirdButtonTimeout" "750"
    Option "EmulateThirdButtonMoveThreshold" "30"
EndSection

all work ok now, I am writing via BBN/lysmarine

PS: I think we have discussed this before in different thread
thank you

**Reply by mgrouch** [reply]
Thanks. I’ve commented out widevine install line from bbn-hot-fix-install.sh now.

**Comment by mgrouch** [comment]
Outdated discussion given new major release with bookworm
